sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/EncryptionCosmosAsyncContainer.java

```diff
@@ -163,29 +204,66 @@
         EncryptionItemRequestOptions encryptionItemRequestOptions = Utils.as(requestOptions,
             EncryptionItemRequestOptions.class);
 
-        if (encryptionItemRequestOptions != null && encryptionItemRequestOptions.getEncryptionOptions() != null) {
-            Preconditions.checkArgument(partitionKey != null, "partitionKey cannot be null for operations using "
-                + "EncryptionContainer.");
+        if (encryptionItemRequestOptions == null || encryptionItemRequestOptions.getEncryptionOptions() == null) {
+            return container.createItem(item, partitionKey, requestOptions);
+        }
+
```

Comment:
NIT remove one line - also below after the precondition check

Reply:
thanks. next commit will fix.